drivers/adc/adc_stm32.c

```diff
@@ -801,6 +796,100 @@ static void adc_stm32_setup_speed(const struct device *dev, uint8_t id,
 #endif
 }
 
+static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_id)
+{
+#if defined(CONFIG_SOC_SERIES_STM32G4X) || \
+	defined(CONFIG_SOC_SERIES_STM32F3X) || \
+	defined(CONFIG_SOC_SERIES_STM32F3X) || \
+	defined(CONFIG_SOC_SERIES_STM32F7X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32L5X) || \
+	defined(CONFIG_SOC_SERIES_STM32MP1X)
+	const struct adc_stm32_cfg *config = dev->config;
+	ADC_TypeDef *adc = config->base;
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32G4X)
+	/* In the G4 model, once can access internal temperature on ADCs 1 and 5. */
+	if ((adc == ADC1 && __LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR_ADC1)
+		== channel_id) || (adc == ADC5 &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR_ADC5)
+		== channel_id)) {
```

Comment:
Something as follows would be easier to read
```suggestion
	if ((adc == ADC1 && 
	     __LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR_ADC1) == channel_id) || 
	     (adc == ADC5 &&
		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR_ADC5) == channel_id)) {
```

Reply:
(in github comment, it isn't obvious though)